dvc/repo/index.py

```diff
@@ -457,6 +457,9 @@ def data(self) -> "Dict[str, DataIndex]":
             if not out.use_cache:
                 continue
 
+            if not out.is_in_repo:
+                continue
```

Comment:
Obviously thought about making `use_cache` automatically check this and return `False`, but I think an explicit approach is better here for now, feels like cached ext outs might make a come back. Might revisit in a followup.